cli/fathom_web/commands/train.py

```diff
@@ -46,18 +51,53 @@ def learn(learning_rate, iterations, x, y, confidence_threshold, num_prunes, val
                         previous_validation_loss = validation_loss
                         previous_model = model.state_dict()
                 writer.add_scalar('validation_loss', validation_loss, t)
-            accuracy, _, _ = accuracy_per_tag(y, y_pred, confidence_threshold, num_prunes)
+            accuracy, _, _ = accuracy_per_tag(y, y_pred, cutoff=0.5, num_prunes=num_prunes)
             writer.add_scalar('training_accuracy_per_tag', accuracy, t)
             optimizer.zero_grad()  # Zero the gradients.
             loss.backward()  # Compute gradients.
             optimizer.step()
     if stopped_early:
         print(f'Stopping early at iteration {t}, just before validation error rose.')
 
+    optimal_cutoff = find_optimal_cutoff(y, y_pred, num_prunes)
     # Horizontal axis is what confidence. Vertical is how many samples were that confidence.
     writer.add_histogram('confidence', confidences(model, x), t)
     writer.close()
-    return model
+    return model, optimal_cutoff
+
+
+def get_possible_cutoffs(y_pred):
+    """Using y_pred gets the sigmoid values, rounds the values and then gets the unique list.
+    This will reduce the number of cutoffs to be evaluated."""
+    with torch.no_grad():
+        flattened = y_pred.sigmoid().numpy().flatten()
+        cutoffs = np.sort(flattened)
+        if len(cutoffs) > 1:
+            cutoffs = np.unique([round((current + next) / 2, CUTOFF_DECIMAL_PLACES) for current, next in pairwise(cutoffs)])
+        return cutoffs
+
+
+def find_optimal_cutoff(y, y_pred, num_prunes):
+    """Evaluates possible cutoff values using accuracy as the metric.
+    If more than 1 cutoff gives the highest accuracy the midpoint
+    between the min and max cutoff is used."""
+    max_accuracy = 0
+    optimal_cutoffs = []
+
+    possible_cutoffs = get_possible_cutoffs(y_pred)
+    for _, test_cutoff in enumerate(possible_cutoffs):
+        accuracy, _, _ = accuracy_per_tag(y, y_pred, test_cutoff, num_prunes)
+        if accuracy == max_accuracy:
+            optimal_cutoffs.append(test_cutoff)
+        if accuracy > max_accuracy:
+            optimal_cutoffs = [test_cutoff]
+            max_accuracy = accuracy
+
+    # Should always have at least 1 cutoff unless something has really gone sideways.
+    optimal_cutoff = optimal_cutoffs[0]
+    if len(optimal_cutoffs) > 1:
+        optimal_cutoff = round((optimal_cutoffs[0] + optimal_cutoffs[len(optimal_cutoffs) - 1]) / 2, CUTOFF_DECIMAL_PLACES)
```

Comment:
I've decided to go with calculating the mean of the possible cutoffs and selecting the value from the possible cutoffs that is closest to the mean.

Reply:
I assume when you say "possible", you mean "optimal". (Correct me if necessary.) Sounds fine. Since it's slightly more complicated to implement than just picking the middle item of the array, be sure to leave a comment explaining why you're doing it that way. It doesn't have to prove anything; if you're just following a hunch, say why you have the hunch and leave it at that. Thanks! :-D

Reply:
Yes "possible" should read "optimal".  Will do wrt code comments.